valhalla/thor/timedep.h

```diff
@@ -248,6 +246,9 @@ class TimeDepReverse : public TimeDepForward {
   // bidirectional edge label structure.
   std::vector<sif::BDEdgeLabel> edgelabels_rev_;
 
+  // Adjacency list - approximate double bucket sort
+  std::shared_ptr<baldr::DoubleBucketQueue<sif::BDEdgeLabel>> adjacencylist_rev_;
```

Comment:
`adjacencylist_` above got moved to private, should this one also be made private?

Reply:
i moved that one to private so that the reverse class couldnt see it. this is important because before this change forward and reverse algorithms shared the same adjacencylist. with the template they can no longer do that because they use 2 different types of labels (EdgeLabel and BDEdgeLabel). if anything i should probably make edgelabels_ private in the forward algorithm to prevent confusion there. basically the only thing they share now is edtge status.

i think its all not a big deal at the moment. my ultimate goal is to combine the 3 things that make up a "Queue/Visited" in the conventional sense of the words with respect to classical routing algorithms. we should come up with a decent name for a class that handles those functions then just do the refactor imho